### PR TITLE
Pagopa 848 sviluppo migrazione receipt table da storage account a cosmos db

### PR DESCRIPTION
**Pull request by FedericoRuzzier**
#### List of Changes
Changing the interaction with the db, using a cosmos table instead of azure table storage

#### Motivation and Context
This change was required due to the slow performance of the table storage

#### How Has This Been Tested?
This has been tested manually

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by FedericoRuzzier** [comment]
Unit tests do not work, there are errors with the cosmos emulator

File: src/main/java/it/gov/pagopa/payments/service/PaymentsService.java [hunk 1]
```diff
@@ -1,290 +1,168 @@
...
+                throw new AppException(
+                        AppError.UNPROCESSABLE_RECEIPT,
+                        paymentOption.getStatus(),
+                        tableEntity.getPartitionKey(),
+                        tableEntity.getRowKey());
+            }
+            // if no exception is raised the status on GPD is correctly in PAID -> for congruence update
+            // receipt status
+            tableEntity.addProperty(STATUS_PROPERTY, Status.PAID.name());
+            tableClient.updateEntity(tableEntity);
```
Comment (AngeloCaporaso) [inline src/main/java/it/gov/pagopa/payments/service/PaymentsService.java:98]: missing `try {} catch () {}`

File: src/main/java/it/gov/pagopa/payments/service/PaymentsService.java [hunk 2]
```diff
@@ -1,290 +1,168 @@
...
-          CloudStorageAccount.parse(storageConnectionString)
-              .createCloudTableClient()
-              .getTableReference(receiptsTable);
+    public List<ReceiptEntity> getGPDCheckedReceiptsList(List<ReceiptEntity> result, TableClient tableClient) {
+        // for all the receipts in the azure table, only those that have been already PAID status or are
+        // in PAID status on GPD are returned
+        List<ReceiptEntity> checkedReceipts = new ArrayList<>();
+        for (ReceiptEntity re : result) {
+            try {
+                this.checkGPDDebtPosStatus(new TableEntity(re.getOrganizationFiscalCode(), re.getIuv()), tableClient);
```
Comment (AngeloCaporaso) [inline src/main/java/it/gov/pagopa/payments/service/PaymentsService.java:108]: extend ReceiptTable with TableEntity could impact on this chunk of code?

File: src/main/java/it/gov/pagopa/payments/utils/CommonUtil.java [hunk 3]
```diff
@@ -32,6 +32,19 @@ public PageInfo buildPageInfo(PaymentsResult<ReceiptEntity> receipts) {
...
+  /**
+   * @param receipts Page returned from the database
+   * @return return the page info
+   */
+  public PageInfo buildPageInfoCosmos(PaymentsResult<ReceiptEntity> receipts) {
+    return PageInfo.builder()
+            .limit(receipts.getPageSize())
+            .morePages(receipts.isHasMoreResults())
+            .itemsFound(receipts.getLength())
+            .page(receipts.getCurrentPageNumber())
```
Comment (AngeloCaporaso) [inline src/main/java/it/gov/pagopa/payments/utils/CommonUtil.java:44]: maybe page must be removed or is necessary to explore further this solution

**Review by AngeloCaporaso — COMMENTED** [review]
Great work 🥇 

some little changes are required.

File: src/main/java/it/gov/pagopa/payments/model/ReceiptModelResponse.java [hunk 4]
```diff
@@ -13,5 +13,6 @@ public class ReceiptModelResponse {
 
   private String organizationFiscalCode;
   private String iuv;
-  private String debtorFiscalCode;
+  private String debtor;
+  private String paymentDateTime;
```
[the remark this answers is not in the document]
Comment (FedericoRuzzier) [reply src/main/java/it/gov/pagopa/payments/model/ReceiptModelResponse.java:17]: Declared in entity, set in createEntity in partnerService, retrieved in paymentService

**Review by AngeloCaporaso — APPROVED** [review]

File: helm/values-prod.yaml [hunk 5]
```diff
@@ -4,7 +4,7 @@ microservice-chart:
   fullnameOverride: ""
   image:
     repository: pagopapcommonacr.azurecr.io/pagopagpdpayments
-    tag: "0.0.17"
+    tag: "0.0.18"
```
Comment (jacopocarlini) [inline helm/values-prod.yaml:7]: AZURE_TABLES_TABLENAME and AZURE_TABLES_CONNECTION_STRING are missing?

File: src/main/resources/application-local.properties [hunk 6]
```diff
@@ -16,6 +16,15 @@ service.apiconfig.host=http://localhost:8084/apiconfig/api/v1
 payments.sa.connection=DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;TableEndpoint=http://127.0.0.1:10002/devstoreaccount1;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;
 receipts.table=receipts
 
+#cosmos data
+azure.cosmos.uri=placeholder
```
Comment (jacopocarlini) [inline src/main/resources/application-local.properties:20]: put the values of cosmos emulator to run the project properly in local

**Pull request merged**